src/components/CountryCode.tsx

```diff
@@ -0,0 +1,174 @@
+import Image from "next/image";
+import { AngleDownUp } from "../newHomepage/assets/Icons";
+import { Dispatch, SetStateAction, useEffect, useRef, useState } from "react";
+
+interface CountryProps {
+    name: {
+        common: string;
+    };
+    cca2: string;
+    idd?: {
+        root: string;
+        suffixes: string[];
+    };
+    flags: {
+        png: string;
+    };
+}
+
+interface CountryListProps {
+    CCInputWidth: string;
+    onSetCountryCode: Dispatch<SetStateAction<string>>;
+}
+
+function CountryCodeList({ CCInputWidth, onSetCountryCode }: CountryListProps) {
+    const [countries, setCountries] = useState<CountryProps[]>([]);
+    const [isClicked, setIsClicked] = useState<boolean>(false);
+    const [selectedCountry, setSelectedCountry] = useState<string>("");
+    const [query, setQuery] = useState<string>("");
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+    useEffect(() => {
+        if (isClicked && inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, [isClicked]);
+
+    useEffect(() => {
+        async function getCountries() {
+            try {
+                const res = await fetch(`https://restcountries.com/v3.1/all`);
+                const data = await res.json();
+                setCountries(data);
+            } catch (error) {
+                console.log("data not fetched");
+            }
+        }
+        getCountries();
+    }, []);
+
+    useEffect(() => {
+        function handleClickOutside(event: MouseEvent) {
+            if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+                setIsClicked(false);
+            }
+        }
+        isClicked
```

Comment:
This check can be done inside `handleClickOutside`. There's no need to add and remove the event listener multiple times